### ISSUE TITLE
[FP] `UnnecessaryCastingInspection` when casting result of division of integers to integer

### ISSUE THREAD

**Issue by paul-sonnenschein** [kalessil/phpinspectionsea#2022]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 6.4.0 |
| Language level | 8.5 |

Current behaviour
--
An `UnnecessaryCastingInspection` is shown when casting a division result (a double) to an integer.
A minimal example triggering the false positive would be:
```php
<?php
echo (int) (12/10) . PHP_EOL
```

An example also demonstrating why the inspection is wrong would be:
```php
<?php
set_error_handler(static function(int $errno, string $errstr) {
    echo 'Error ' . $errstr . PHP_EOL;
});

echo 'Division result type: ' . gettype(12 / 10) . PHP_EOL;
echo 'Test 1A: ' . (12 / 10) % 10 . PHP_EOL;
echo 'Test 1B: ' . (12 / 10) . PHP_EOL;
echo 'Test 2A: ' . (int) (12 / 10) % 10 . PHP_EOL;
echo 'Test 2B: ' . (int) (12 / 10) . PHP_EOL;
```

Which outputs, demonstrating that the cast is not unnecessary:
```
Division result type: double
Error Implicit conversion from float 1.2 to int loses precision
Test 1A: 1
Test 1B: 1.2
Test 2A: 1
Test 2B: 1
```

Expected behaviour
--
The inspection does not trigger when casting the result of dividing two integers (a floating point number) into an integer.

Environment details
---

PhpStorm 2026.2.1
